virtual_ecosystem/models/plants/plants_model.py

```diff
@@ -317,28 +403,29 @@ def set_canopy_absorption(self, time_index: int) -> None:
         canopy for a particular time index and uses the ``layer_fapar`` data calculated
         by the canopy model to estimate the irradiance absorbed by each layer and the
         remaining irradiance at ground level.
-        """
 
-        # TODO: With the full canopy model, this could be partitioned into sunspots and
-        #       shade.
+        TODO:
+          - With the full canopy model, this could be partitioned into sunspots
+            and shade.
+          - At the moment, we're only looking at PPFD and need to switch to SWDown
+            `#721 <https://github.com/ImperialCollegeLondon/virtual_ecosystem/issues/721>`_
+        """  # noqa: D405 temporary section
 
         # Extract a PPFD time slice
         canopy_top_ppfd = (
             self.data["photosynthetic_photon_flux_density"]
             .isel(time_index=time_index)
-            .data
+            .to_numpy()
         )
 
         # Calculate the fate of PPFD through the layers
-        absorbed_irradiance = canopy_top_ppfd * self.data["layer_fapar"].data
-        ground_irradiance = canopy_top_ppfd - np.nansum(absorbed_irradiance, axis=0)
+        absorbed_irradiance = self.data["layer_fapar"] * canopy_top_ppfd
+        # Add the remaining irradiance at the surface layer level
+        absorbed_irradiance[self.layer_structure.index_surface] = (
```

Comment:
This is an interesting point to discuss, the abiotic model does the same calculation, I was not aware that this is already part of the `canopy_absorption` variable. From a physical perspective I'm not sure if that makes sense to allocate the remaining energy to the surface layer. We could make the assumption that this is absorbed by the topsoil layer or store it in a separate variable or not at all? What do you think?

Reply:
`# Net radiation topsoil, [W m-2]
        net_radiation_soil = energy_balance.calculate_net_radiation(
            incoming_radiation=data["topofcanopy_radiation"].isel(time_index=time_index).to_numpy(),
            absorbed_radiation=np.nansum(data["absorbed_radiation"].to_numpy(), axis=0),
            longwave_emission=longwave_emission_soil,
            albedo=abiotic_constants.surface_albedo,
        )`

Reply:
That's a mistake on my part - the topsoil layer is clearly the right place to dump it. I think it is cleaner if my array identify the fate of all the radiation through the canopy, partly because at some point animals may be interested in sunflecks and the like, which will become part of this.

Reply:
I've moved those values down into the topsoil layer.

Reply:
Is the output variable still called `canopy_absorption`? This could be a bit misleading, would shortwave absorption be more precise?